src/Tools/dotnet-openapi/src/Commands/AddFileCommand.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace Microsoft.DotNet.OpenApi.Commands
+{
+    internal class AddFileCommand : BaseCommand
+    {
+        private const string CommandName = "file";
+
+        public AddFileCommand(AddCommand parent)
+            : base(parent, CommandName)
+        {
+            _classNameOpt = Option("-c|--class-name", "The name of the class to be generated", CommandOptionType.SingleValue);
+            _outputFileOpt = Option("-o|--output-file", "The name of the file to output the swagger file to", CommandOptionType.SingleValue);
```

Comment:
I am not sure it's necessary to handle all metadata options `@(OpenApiReference)` supports. I suggest supporting only `--download-dir` (as speced) and perhaps `--namespace` i.e. neither of these.